Libraries/Network/RCTNetworking.mm

```diff
@@ -417,13 +456,23 @@ - (void)sendRequest:(NSURLRequest *)request
   RCTURLRequestProgressBlock downloadProgressBlock = nil;
   if (incrementalUpdates) {
     if ([responseType isEqualToString:@"text"]) {
+			
+			//We need this to carry over bytes, which could not be decoded into text (such as broken UTF-8 characters).
+			//The incremental data block holds the ownership of this object, and will be released upon release of the block.
+			NSMutableData* incrementalDataCarryStorage = [NSMutableData new];
+			
       incrementalDataBlock = ^(NSData *data, int64_t progress, int64_t total) {
-        NSString *responseString = [RCTNetworking decodeTextData:data fromResponse:task.response];
+				NSUInteger initialCarryLength = incrementalDataCarryStorage.length;
+				
+				NSString *responseString = [RCTNetworking decodeTextData:data fromResponse:task.response withCarryStorage:incrementalDataCarryStorage];
         if (!responseString) {
           RCTLogWarn(@"Received data was not a string, or was not a recognised encoding.");
           return;
         }
-        NSArray<id> *responseJSON = @[task.requestID, responseString, @(progress), @(total)];
+				
+				//Update progress to include the previous carry length and reduce the current carry length.
+				NSArray<id> *responseJSON = @[task.requestID, responseString, @(progress + initialCarryLength - incrementalDataCarryStorage.length), @(total)];
+				
```

Comment:
a few code styling nits to address:
* comment lines: add space after `//`
* please make sure indentation is 2 spaces consistently
* for long method call, could you split them into multiple lines (e.g. line 474, 467, 368)?
* `if(` --> `if (`, `while(` --> `while (` (space in between)
* curly braces for `if`, `else`, and `while` should start at the same line: `if (...) {`